### ISSUE TITLE
Port to TypeScript

### ISSUE THREAD

**Issue by chuwy** [snowplow/iglu-javascript-client#15]
Hey @mhadam, @lhache! 

@Volland is keen to explore TypeScript for this Iglu JS Client and wondering if we have any particular internal conventions or common tslint config?

**Comment by lwih** [comment]
I'm using TypeScript as well but as I'm not much of a man of conventions, I'm fairly loose there. Though, I'd be happy to see what you're going to set there @Volland.

**Comment by chuwy** [comment]
Makes sense. I don't think we have very big TypeScript code estate, but I'd very happy if it will start to grow.
